source/models/table.py

```diff
@@ -36,26 +37,69 @@ def get_path(self):
         
     def get_name(self):
         return self.__table_name
-
+    
     def set(self, row):
+        self.__colomns_name_validate(row = row)
         row_obj = Row(table = self, value = row)
         if row_obj.row_exists() and self.table_metadata.enable_overwrite == "false":
             raise RowExists(message = "this row is already exists and can't overwrite")
         elif row_obj.row_exists():
-            old_row = Row.load_by_primary_key(table = self, primary_key = row_obj.get_primary_key())
+            old_row = Row(table = self).load_by_primary_key(table=self, primary_key = row_obj.primary_key)
```

Comment:
revert this line, 
The old line was right 
`            old_row = Row.load_by_primary_key(table = self, primary_key = row_obj.get_primary_key())
`